servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -0,0 +1,571 @@
+    super.start(isLeader);
+    if (isLeaderMode()) {
+      mLostWorkerDetectionService =
+          getExecutorService().submit(
+              new HeartbeatThread("Lost worker detection service",
+                  new LostWorkerDetectionHeartbeatExecutor(), mTachyonConf
+                      .getInt(Constants.MASTER_HEARTBEAT_INTERVAL_MS)));
+    }
+  }
+
+  @Override
+  public void stop() throws IOException {
+    super.stop();
+    if (isLeaderMode()) {
+      if (mLostWorkerDetectionService != null) {
+        mLostWorkerDetectionService.cancel(true);
+      }
+    }
+  }
+
+  /**
+   * Gets the number of workers.
+   *
+   * @return the number of workers
+   */
+  public int getWorkerCount() {
+    synchronized (mWorkers) {
+      return mWorkers.size();
+    }
+  }
+
+  public List<WorkerInfo> getWorkerInfoList() {
+    List<WorkerInfo> workerInfoList = new ArrayList<WorkerInfo>(mWorkers.size());
+    synchronized (mWorkers) {
+      for (MasterWorkerInfo masterWorkerInfo : mWorkers) {
+        workerInfoList.add(masterWorkerInfo.generateClientWorkerInfo());
+      }
+    }
+    return workerInfoList;
+  }
+
+  public long getCapacityBytes() {
+    long ret = 0;
+    synchronized (mWorkers) {
+      for (MasterWorkerInfo worker : mWorkers) {
+        ret += worker.getCapacityBytes();
+      }
+    }
+    return ret;
+  }
+
+  public long getUsedBytes() {
+    long ret = 0;
+    synchronized (mWorkers) {
+      for (MasterWorkerInfo worker : mWorkers) {
+        ret += worker.getUsedBytes();
+      }
+    }
+    return ret;
+  }
+
+  public Set<Long> getLostBlocks() {
+    return Collections.unmodifiableSet(mLostBlocks);
+  }
+
+  /**
+   * Gets info about the lost workers
+   *
+   * @return a list of worker info
+   */
+  public List<WorkerInfo> getLostWorkersInfo() {
+    List<WorkerInfo> ret = new ArrayList<WorkerInfo>();
+
+    for (MasterWorkerInfo worker : mLostWorkers) {
+      ret.add(worker.generateClientWorkerInfo());
+    }
+    return ret;
+  }
+
+  public void removeBlocks(List<Long> blockIds) {
+    for (long blockId : blockIds) {
+      MasterBlockInfo masterBlockInfo = mBlocks.get(blockId);
+      if (masterBlockInfo == null) {
+        return;
+      }
+      for (long workerId : masterBlockInfo.getWorkers()) {
+        masterBlockInfo.removeWorker(workerId);
+        MasterWorkerInfo worker = mWorkers.getFirstByField(mIdIndex, workerId);
+        if (worker != null) {
+          worker.updateToRemovedBlock(true, blockId);
+        }
+      }
+      // remove from lost blocks
+      mLostBlocks.remove(blockId);
+    }
+  }
+
+  @Override
+  public long getNewContainerId() {
+    synchronized (mBlockIdGenerator) {
+      long containerId = mBlockIdGenerator.getNewBlockContainerId();
+      // Write id generator state to the journal.
+      writeJournalEntry(new BlockIdGeneratorEntry(containerId));
+      flushJournal();
+      return containerId;
+    }
+  }
+
+  /**
+   * Commit a block on a specific worker.
```

Comment:
I think the javadoc should be more like:
notified by a worker that a block is committed a specified tier ?